### PR TITLE
feat: add fromSubscriptionName to purchase and refund events

### PR DESCRIPTION
**Pull request by antsgar**

### CODE REVIEW

File: packages/domain-events/src/Domain/Event/SubscriptionPurchasedEventPayload.ts [hunk 1]
```diff
@@ -2,7 +2,8 @@ import { SubscriptionName } from '@standardnotes/auth'
 
 export interface SubscriptionPurchasedEventPayload {
   userEmail: string
-  subscriptionName: SubscriptionName
+  fromSubscriptionName: SubscriptionName
```
Comment (karolsojko) [inline packages/domain-events/src/Domain/Event/SubscriptionPurchasedEventPayload.ts:5]: from can be undefined right ? - purchasing a subscription from not having one
```suggestion
  fromSubscriptionName?: SubscriptionName
```

File: packages/domain-events/src/Domain/Event/SubscriptionRefundedEventPayload.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { SubscriptionName } from '@standardnotes/auth'
 
 export interface SubscriptionRefundedEventPayload {
   userEmail: string
-  subscriptionName: SubscriptionName
+  fromSubscriptionName: SubscriptionName
+  toSubscriptionName: SubscriptionName
```
Comment (karolsojko) [inline packages/domain-events/src/Domain/Event/SubscriptionRefundedEventPayload.ts:6]: hmm if refunding we are cancelling a subscription so to say ... so no 'toSubscriptio` is needed right?

File: packages/domain-events/src/Domain/Event/SubscriptionPurchasedEventPayload.ts (hunk 1 above)
Comment (antsgar) [reply packages/domain-events/src/Domain/Event/SubscriptionPurchasedEventPayload.ts:5]: The fromSubscriptionName in that case should be CorePlan, correct? That way we would know we need to remove the CoreUser role from the user

File: packages/domain-events/src/Domain/Event/SubscriptionRefundedEventPayload.ts (hunk 2 above)
Comment (antsgar) [reply packages/domain-events/src/Domain/Event/SubscriptionRefundedEventPayload.ts:6]: Same as above, the toSubscriptionName should be CorePlan. Does that make sense?

File: packages/domain-events/src/Domain/Event/SubscriptionPurchasedEventPayload.ts (hunk 1 above)
Comment (karolsojko) [reply packages/domain-events/src/Domain/Event/SubscriptionPurchasedEventPayload.ts:5]: hmm not really ... a basic user: https://github.com/standardnotes/snjs/blob/master/packages/auth/src/Domain/Role/RoleName.ts#L2

can buy a core plan: https://github.com/standardnotes/snjs/blob/master/packages/auth/src/Domain/Subscription/SubscriptionName.ts#L2

so it means that his subscription was null until now.

File: packages/domain-events/src/Domain/Event/SubscriptionRefundedEventPayload.ts (hunk 2 above)
Comment (karolsojko) [reply packages/domain-events/src/Domain/Event/SubscriptionRefundedEventPayload.ts:6]: When someone cancels a core subscription they can get back to a basic user with no subscription

File: packages/domain-events/src/Domain/Event/SubscriptionPurchasedEventPayload.ts (hunk 1 above)
Comment (antsgar) [reply packages/domain-events/src/Domain/Event/SubscriptionPurchasedEventPayload.ts:5]: Ah thanks, that clears everything out! For some reason I was confusing Core with Basic thinking they were the same thing. Will update this

File: packages/domain-events/src/Domain/Event/SubscriptionRefundedEventPayload.ts (hunk 2 above)
Comment (antsgar) [reply packages/domain-events/src/Domain/Event/SubscriptionRefundedEventPayload.ts]: @karolsojko come to think of it, it's the same case as above but reversed, so should be optional I think? A user can have for example a Plus subscription, buy a Pro subscription and then refund it. In that case, the user would go from Pro to Plus. Is that correct?

Comment (antsgar) [reply packages/domain-events/src/Domain/Event/SubscriptionRefundedEventPayload.ts]: @karolsojko actually I just realized we may not need this at all, how about we just add the new user role when a subscription is purchased and remove it if it's refunded? That way if a user doesn't have a subscription and a Pro plan is purchased for example, the user would have both the BasicUser and ProUser roles, but I think that'd be okay. Wdyt?

Comment (antsgar) [reply packages/domain-events/src/Domain/Event/SubscriptionRefundedEventPayload.ts]: Made the described change on the auth PR so will close this for now, will re-open if necessary 🙂

**Pull request closed without merging**